data-prepper-plugins/dynamodb-source-coordination-store/src/test/java/org/opensearch/dataprepper/plugins/sourcecoordinator/dynamodb/DynamoDbClientWrapperTest.java

```diff
@@ -175,7 +216,51 @@ void tryCreateTableThrows_runtime_exception_when_waiter_returns_empty_describe_r
 
             given(waiterResponse.matched()).willReturn(response);
 
-            assertThrows(RuntimeException.class, () -> objectUnderTest.tryCreateTable(tableName, provisionedThroughput));
+            assertThrows(RuntimeException.class, () -> objectUnderTest.initializeTable(dynamoStoreSettings, provisionedThroughput));
+        }
+
+        verify(table, never()).createTable(any(CreateTableEnhancedRequest.class));
+    }
+
+    @Test
+    void skip_table_creation_true_does_not_attempt_to_create_the_table_or_enable_ttl() {
+        final DynamoDbClientWrapper objectUnderTest = createObjectUnderTest();
+
+        final String tableName = UUID.randomUUID().toString();
+        final ProvisionedThroughput provisionedThroughput = mock(ProvisionedThroughput.class);
+
+        given(dynamoStoreSettings.getTableName()).willReturn(tableName);
+        given(dynamoStoreSettings.skipTableCreation()).willReturn(true);
+        given(dynamoStoreSettings.getTtl()).willReturn(Duration.ofSeconds(30));
+
+        final DynamoDbTable<DynamoDbSourcePartitionItem> table = mock(DynamoDbTable.class);
+        given(dynamoDbEnhancedClient.table(eq(tableName), any(BeanTableSchema.class))).willReturn(table);
+
+        try (MockedStatic<DynamoDbWaiter> dynamoDbWaiterMockedStatic = mockStatic(DynamoDbWaiter.class)) {
+            final DynamoDbWaiter dynamoDbWaiter = mock(DynamoDbWaiter.class);
+            dynamoDbWaiterMockedStatic.when(DynamoDbWaiter::create).thenReturn(dynamoDbWaiter);
+            final WaiterResponse waiterResponse = mock(WaiterResponse.class);
```

Comment:
I think you could move the rest of the mocking to outside this try-with-resources block. It can help make the code a bit more readable.

Reply:
I wasn't able to call the initialize method outside of the try-with-resources block. But I see what you mean I will move the mock objects to be above the try-with-resources where I can